### DISCUSSION TITLE
Sprint 2 Retrospective

### DISCUSSION THREAD

**Discussion by DJ-CO** [annalieseschropp/checkers#94] [Sprint Retrospective]
Greetings everyone. Please answer the following questions to help us improve for this coming sprint.

1. What went well in this sprint?
2. What could be improved upon in the next sprint?
3. What have you learned?
4. Anything else you want to say.

**Comment by DJ-CO** [comment]
1) What went well in this sprint?
* We smoked the other team, accomplishing an incredible amount this sprint.

2) What could be improved upon in the next sprint?
* Git management was a critical issue this sprint:
  * Branches with non-trivial merge conflicts
  * Extraneous files committed
  * Commits destroying previous progress

3) What have you learned?
* Dennis can be wrong

4) Anything else you want to say.
* Nothing really.

**Comment by imckechn** [comment]
1. What went well in this sprint?

- Created a bunch of screens and features that made the game playable and kept the game looking good at the same time. We kept to the pastel theme and that has a very nice appeal for the game.

2. What could be improved upon in the next sprint?

- I was a little slow getting my part up and running, it took until the very last day. I would like to be more efficient and have it in before the last day. 

3. What have you learned?

- I learned how to merge Unity Scenes thanks to my team. 

4. Anything else you want to say.

- None

**Comment by Sahibjeet-Babbrah** [comment]
1. What went well in this sprint?
        - We accomplished an incredible amount this sprint and smoked the other team.
        - We completed all that needed to be done and can now just focus on polishing the game.
2. What could be improved upon in the next sprint?
        - We had some git problems this sprint which need to be ironed out.
                - Commits destroying previous commits
                - Branches containing excess files
3. What have you learned?
        - We shouldn't try to parallelize something that can't be parallelized.
4. Anything else you want to say?
        - NULL

**Comment by annalieseschropp** [comment]
Annie
1. What went well in this sprint?
    * “We smoked the other team, accomplishing an incredible amount this sprint.” - D.J.

2. What could be improved upon in the next sprint?
    * I think similarly to the last sprint, our git practices need to be improved upon. Such as branch naming convention to follow the syntax of #.#_name_of_task, branching off of the dev branch, frequently committing work with detailed messages of exactly what was committed, ensuring commits are “bite-sized” pieces of progress, only committing necessary files, ensuring the code that is committed won’t cause merge conflicts or correct the code to fix them, ensure the code is thoroughly tested before submitting a PR, thoroughly documented PR comments (see first PR for example), and adding reviewers to test the work done, etc. 

3. What have you learned?
    * Parallelization is not always the solution. 

4. Anything else you want to say.
    * N/A
